lucille-core/src/main/java/com/kmwllc/lucille/util/ElasticsearchUtils.java

```diff
@@ -125,4 +129,79 @@ public static boolean getAllowInvalidCert(Config config) {
     }
     return false;
   }
+
+  public static class ElasticJoinData {
+
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    private String joinFieldName;
+    private boolean isChild;
+    private String parentName;
+    private String childName;
+    private String parentDocumentIdSource;
+
+    public void addJoinData(Document doc) {
```

Comment:
a method starting with "add" usually adds the argument to the object that the method is called on
in this case, we're copying from the object onto the argument
suggest renaming this populateJoinData() or copyJoinDataToDocument()